CRM/Utils/SQL/BaseParamQuery.php

```diff
@@ -184,6 +184,7 @@ public function param($keys, $value = NULL) {
    *
    * @return bool
    */
+  #[ReturnTypeWillChange]
```

Comment:
that is super interesting - it does feel like this particular one should be reliably a bool though?